libraries/esp8266/examples/LowPowerDemo/README.md

```diff
@@ -0,0 +1,106 @@
+
+### Test 6 - Deep Sleep, wake with RF_DEFAULT
+
+In Deep Sleep almost everything is turned off, and the chip draws ~20 uA.  If you have D0/GPIO16 connected to RST, you can use the RTC timer to wake the chip up at a timed interval.  You can also wake it with an external RESET.  Waking with RF_DEFAULT means it will do an RFCAL if it needs to.  These first two Deep Sleep tests use the standard Deep Sleep function, so the WiFi connection is closed and the modem turned off, which takes about 270mS before Deep Sleep begins.  Deep Sleep ends with a RESET, and the boot time after that is around 120mS.  Any Deep Sleep less than 2 seconds is wasting power due to the boot time and modem shut-off, and Forced Light Sleep will be a better choice.
+
+### Test 7 - Deep Sleep, wake with RFCAL
+
+Identical to the test above, but the modem does a power calibration when booting.  In normal use, I'd likely do WAKE_RF_DEFAULT instead to save the extra RFCAL power burst if it's not needed.
+
+### Test 8 - Deep Sleep Instant, wake with NO_RFCAL
+
+This variation doesn't do an automatic RF calibration, so power requirements will be slightly less.  Additionally, *most* of the time it immediately goes into Deep Sleep without turning off the modem (that's the INSTANT part).  There's another bug in SDK 2, and I see both SDK functions and WiFi-class functions occasionally do a modem shut-down before Deep Sleep; it's not always Instant.  When it doesn't do the modem shut-down it saves an extra 270mS of power.
+
+### Test 9 - Deep Sleep Instant, wake with RF_DISABLED
+
+This last variation also uses Deep Sleep Instant, but it wakes up with the modem disabled so current after Deep Sleep is only 15 mA.  You'll have to turn the modem back on in your program when you want to use WiFi again.  Each of the 4 WAKE styles has their own use, depending on what you need.
+
+---
+
+All of the Deep Sleep modes end with a RESET, so you must re-initialize everything.  You can store *some* information in the RTC memory to survive a Deep Sleep reset, which I've done in this demo to illustrate it.  See the **RTCUserMemory** example for more on this feature.  
+
+The maximum Deep Sleep interval is 71.58 minutes (2^32 -1 microseconds).
+
+If you need a longer sleep period than that, you can pass zero as the time variable to Deep Sleep and it turns off the RTC.  The only way to wake it at that point is an external RESET; D0 can't do it.  Both Forced Light Sleep and Deep Sleep(0) are woken by an external signal, so short delays under 2 seconds are more efficient with Forced Light Sleep, and longer delays are more power efficient with Deep Sleep.
+
+
+---
+
+### Lower Power without the WiFi library:
+
+If all you want to do is reduce power for a sketch that doesn't need WiFi, add these SDK 2 functions to your code:
+```c
+  wifi_station_disconnect();
```

Comment:
I didn't see the other example (doing the same thing).  I haven't looked at any examples that weren't of immediate interest.  I saw several people on forums and elsewhere attempting this and failing, instead.  That's why I added it to the README.  You can include that chunk anywhere, all it does it turn off the modem.

Reply:
There's no timeout on the WiFi connection, and none on the pushbutton waits.  Those are the only two waits I know of offhand.  If it never connects, turn on the debug option and see why the WiFi failed.  As far as _never pushing the button_ goes, ummm... how simple ARE kids nowadays?  The bottom line on the serial monitor says "press the button to continue" each time it stops.  Half of the tests need an active WiFi connection, so I don't really want to fail into the code with nothing more than a warning on the monitor that all hell has broken loose on WiFi so we're running half-functional.  Maybe I should be more explicit that it needs an active WiFi connection to run the tests... I never tried without here.

Reply:
I just forced a WiFi fail and continued on into the code: both of the Automatic tests fail as there's no WiFi connection: neither one ever went into low-power mode.  If I put in a timeout on WiFi, I'll need to add additional checks in to skip those two tests in the event of WiFi failure.  The other tests ran OK.  Aren't 99% of people buying the ESP8266 because they want WiFi?

Reply:
OTA is gutted out, and a 30 second timeout on WiFi added.  If it fails connection it skips the two 'Automatic' tests as they won't do anything.  They won't fail or anything, but they won't work without an active connection.  CRC32 integration is the last step I'm working on.

Reply:
Done, hopefully it passes the style check THIS time around.  :stuck_out_tongue_closed_eyes: